translation agile_software_development and devops (#356)
cncf/glossary: content/it/agile_software_development.md

---

> title: Agile Software Development

Reviewer: Please consider translate "Agile Software Development" into Italian, if needed!

Ref: "API Gateway" page in pt-br:
- https://github.com/cncf/glossary/pull/479/files
- https://deploy-preview-479--cncfglossary.netlify.app/pt-br/api_gateway/

![image](https://user-images.githubusercontent.com/46767780/155661288-5cd13792-46b0-4887-b492-9c4cc6336ca2.png)

status: Completed

> category: concept

Reviewer: Please consider translate "concept" into Italian, if needed!

Ref: "API Gateway" page in pt-br:
- https://github.com/cncf/glossary/pull/479/files
- https://deploy-preview-479--cncfglossary.netlify.app/pt-br/api_gateway/

![image](https://user-images.githubusercontent.com/46767780/155661498-18336ac5-2135-4310-ab91-6e64ea74236f.png)

---

## Cos'è
Un insieme di pratiche che enfatizzano i cicli di sviluppo iterativi e i team auto-organizzati. In contrasto con i progetti waterfall (a cascata) in cui il valore è generato solo alla fine del progetto, lo sviluppo agile del software si concentra su una consegna continua e incrementale del valore e sul miglioramento evolutivo del processo stesso.

> ## Quale problema affronta

Reviewer 2: Quali problematiche affronta

> Definire, comunicare e comprendere i requisiti per tutti gli stakeholders (interessati) in un progetto di sviluppo software è molto difficile, se non impossibile. Tuttavia, i clienti vogliono che i loro progetti di sviluppo software siano consegnati in tempo, in buona qualità, in budget e in scopo. Con la sua natura ciclica, lo sviluppo agile del software permette il continuo adattamento dei requisiti e un adattamento più veloce in base a tutte le altre circostanze contrariamente alle strategie waterfall (a cascata). 

Reviewer 3: ```suggestion
Definire, comunicare e comprendere i requisiti per tutti gli stakeholders (interessati) in un progetto di sviluppo software è molto difficile, se non impossibile. Tuttavia, i clienti vogliono che i loro progetti di sviluppo software siano consegnati in tempo, di buona qualità, entro i limiti del budget e dello scopo. Con la sua natura ciclica, lo sviluppo agile del software permette il continuo adattamento dei requisiti e un adattamento più veloce in base a tutte le altre circostanze contrariamente alle strategie waterfall (a cascata). 
```

> Definire, comunicare e comprendere i requisiti per tutti gli stakeholders (interessati) in un progetto di sviluppo software è molto difficile, se non impossibile. Tuttavia, i clienti vogliono che i loro progetti di sviluppo software siano consegnati in tempo, in buona qualità, entro i limiti del budget e dello scopo. Con la sua natura ciclica, lo sviluppo agile del software permette il continuo adattamento dei requisiti e un adattamento più veloce in base a tutte le altre circostanze contrariamente alle strategie waterfall (a cascata). 

Reviewer 4: tra le altre
- una proposta per rimuovere la ripetizione del termine "adattare"
- rimosso `(a cascata)` perché `waterfall` era già presente nel paragrafo precedente
```suggestion
Definire, comunicare e comprendere i requisiti per tutti gli stakeholder (interessati) in un progetto di sviluppo software è molto difficile, se non impossibile. Tuttavia, i clienti vogliono che i loro progetti di sviluppo software siano consegnati in tempo, in buona qualità, entro i limiti del budget e dello scopo. Con la sua natura ciclica, lo sviluppo agile del software consente l'evoluzione continua dei requisiti e un adattamento più veloce in base a tutte le altre circostanze rispetto alle strategie waterfall. 
```

## In che modo aiuta

> Lo sviluppo agile del software contiene tutte le fasi tradizionali delle strategie waterfall (a cascata), come requisiti di ingegneria, pianificazione, implementazione, revisione, test e consegna. La differenza più grande è che l'intero arco di tempo di un progetto software è suddiviso in iterazioni, le cui iterazioni contengono ciascuna di tutte queste fasi. Dopo ogni iterazione, il valore creato può essere rivisto con il cliente e i requisiti possono essere adattati verso l'obiettivo finale. Inoltre il team di sviluppo riflette su quali azioni intraprendere per migliorare il processo stesso.

Reviewer 2: in iterazioni, le cui iterazioni contengono ciascuna di tutte queste fasi.

> ciascuna delle quali contiene tutte queste fasi

> Lo sviluppo agile del software contiene tutte le fasi tradizionali delle strategie waterfall (a cascata), come requisiti di ingegneria, pianificazione, implementazione, revisione, test e consegna. La differenza più grande è che l'intero arco di tempo di un progetto software è suddiviso in iterazioni, ciascuna delle quali contiene tutte queste fasi. Dopo ogni iterazione, il valore creato può essere rivisto con il cliente e i requisiti possono essere adattati verso l'obiettivo finale. Inoltre il team di sviluppo riflette su quali azioni intraprendere per migliorare il processo stesso.

Reviewer 4: ```suggestion
Lo sviluppo agile del software contiene tutte le fasi tradizionali delle strategie waterfall, come requisiti di ingegneria, pianificazione, implementazione, revisione, test e consegna. La differenza sostanziale è che l'intero arco di tempo di un progetto software è suddiviso in iterazioni, ciascuna delle quali contiene tutte queste fasi. Dopo ogni iterazione, il valore creato può essere rivisto con il cliente e i requisiti possono essere adattati verso l'obiettivo finale. Inoltre il team di sviluppo riflette su quali azioni intraprendere per migliorare il processo stesso.
```
